### DISCUSSION TITLE
Set database file in diesel.toml

### DISCUSSION THREAD

**Discussion by jafioti** [diesel-rs/diesel#3191] [Q&A] [answered]
When I have a bunch of different projects all using diesel databases, since they all use the environment variable for the database path, they end up trying to use the same database. This seems like something that should be available to set in the diesel.toml file. 

Is this doable? What is the recommended way to have multiple db files on the same machine? Ideally it wouldn't involve environment variables.

**Comment by weiznich** [comment, marked as answer]
Can you please elaborate your use case a bit more? I cannot follow why you thing this needs to involve environment variables with the current implementation. All places that accept a connection url can be configured with an explicit connection string, non of them is only available by using environment variables. 

Setting the database url in your diesel.toml is not a good solution in my opinion, as you likely do not want to use the same database settings in all locations where the project is used. For local setups a `.env` file + something like the `dotenv` crate is the preferred solution.

**Reply by jafioti** [reply]
I'm talking about for using the cli. Dotenv works ok but it doesn't work on Mac. For these semi small projects I keep the database directly in the project folder, so regardless of the platform it will be in the same place.

**Reply by weiznich** [reply]
Can you elaborate on "dotenv doesn't work on Mac"? That might be something that should be reported upstream and fixed there.

> For these semi small projects I keep the database directly in the project folder, so regardless of the platform it will be in the same place.

This might be the case for your projects, but we cannot assume that generally. As adding an option to the config file invites users to use it + that's a feature where this likely goes wrong in many cases I would prefer not having an option for this.

**Reply by jafioti** [reply]
@weiznich I figured that using .envrc is good enough for my use case since it in fact does work on macos
